contracts/mock-and-referral/CampaignFactory.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "./ReferralCampaign.sol";
+import "@openzeppelin/contracts/proxy/Clones.sol";
+//@notice A Referral Campaign Factory Contract, all referral campaign will be depolyed using this contract
+contract CampaignFactory{
+    //@dev Events
+   event CampaignCreated(address indexed,address);
+    
+    //@dev Referral Campaign Implementation
+    address immutable referralCampaign;
+
+    //@dev Campaign struct with campiagn address and token address
+    struct Campaign{
+        address campaignAddress;
```

Comment:
Can we use terminology consistent with discord?

- rewardsContractAddress
- rewardsAccountAddress
- businessCaseContractAddress or campaignContractAddress
- maxReferralsPerReferer

Also looks like some parameters like refereeAwardPerReferral and referrerAwardPerReferral are missing.